ledger/object/sequence.go

```diff
@@ -0,0 +1,391 @@
+//
+// Copyright 2019 Insolar Technologies GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package object
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+	"sync"
+	"unsafe"
+
+	"github.com/pkg/errors"
+	"github.com/ugorji/go/codec"
+
+	"github.com/insolar/insolar/insolar"
+	"github.com/insolar/insolar/insolar/record"
+	"github.com/insolar/insolar/insolar/utils"
+	"github.com/insolar/insolar/internal/ledger/store"
+)
+
+var (
+	ErrNoSequenceSyncPulses = errors.New("No synchronized pulses")
+)
+
+type SequenceRecordPosition struct {
+	Index       uint32
+	PulseNumber insolar.PulseNumber
+}
+
+func (pos SequenceRecordPosition) String() string {
+	return fmt.Sprintf("(Index: %d, PulseNumber: %d)", pos.Index, pos.PulseNumber)
+}
+
+func (pos SequenceRecordPosition) GreaterThan(other SequenceRecordPosition) bool {
+	return pos.PulseNumber > other.PulseNumber ||
+		(pos.PulseNumber == other.PulseNumber && pos.Index > other.Index)
+}
+
+type SequenceRecord struct {
+	Position SequenceRecordPosition
+	ID       insolar.ID
+	Record   record.MaterialRecord
+}
+
+type SequenceRecordAccessor interface {
```

Comment:
Add comments to the interfaces too, please

Reply:
Added